PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEA.cxx

```diff
@@ -1020,20 +1151,68 @@ Bool_t AliAnalysisTaskEA::FillHistograms(){
    //READ  TRACK AND JET CONTAINERS
    //Container operations   http://alidoc.cern.ch/AliPhysics/master/READMEcontainers.html#emcalContainerIterateTechniques
 
-   fTrkContainerDetLevel = static_cast<AliTrackContainer*> (GetTrackContainer(fMyTrackContainerName.Data())); //reconstructed particle container 
-   fJetContainerDetLevel = static_cast<AliJetContainer*> (GetJetContainer(fMyJetContainerName.Data())); //AKT jet
+   fTrkContainerDetLevel = static_cast<AliTrackContainer*> (GetTrackContainer(fMyTrackContainerName.Data())); //reconstructed track container detector-level 
+   fJetContainerDetLevel = static_cast<AliJetContainer*> (GetJetContainer(fMyJetContainerName.Data())); //AKT jet  container detector-level 
 
 
 
    if(fMC){
-      fParticleContainerPartLevel = GetParticleContainer(fMyParticleContainerName.Data()); //reconstructed particle container 
-      fJetContainerPartLevel      = GetJetContainer(fMyJetParticleContainerName.Data()); //reconstructed particle container 
+      fParticleContainerPartLevel = GetParticleContainer(fMyParticleContainerName.Data()); // particle container particle-level 
+      fJetContainerPartLevel      = GetJetContainer(fMyJetParticleContainerName.Data());   //jet container  particle-level
 
       rhoMC = GetExternalRho(kPartLevel); //estimated backround pt density
    }
+   //_________________________________________________________
+   //                 TT
+
+   for(Int_t i=0; i<fkTTbins; i++){
+      fIndexTTC[i] = -1;
+      fIndexTTH[i] = -1;
+      fIndexTTJ[i] = -1; 
+
+      fIndexTTH_PartLevel[i] = -1;
+      fIndexTTC_PartLevel[i] = -1;
+
+      fTTC[i].resize(0);
+      fTTH[i].resize(0);
+      fTTJ[i].resize(0);
+
+      fTTH_PartLevel[i].resize(0);
+      fTTC_PartLevel[i].resize(0);
+   }
+
+   TLorentzVector ph;
+   for(Int_t i=0; i<fnClusterTTBins; i++){
+      fClusterTT[i] = 0;
+      fClusterTT_PartLevel[i] = 0;
+   }
+
+   for(Int_t i=0; i<fnHadronTTBins; i++){
+      fHadronTT[i] = 0;
+      fHadronTT_PartLevel[i] = 0;
+   }
+
+
    //___________________________________________
    //EVALUATE SINGLE PARTICLE EFFICIENCY + FILL RESPONSE MATRIX
+
+   fMultV0A_PartLevel = 0.;
+   fMultV0C_PartLevel = 0.;
+   fMultV0M_PartLevel = 0.;
+   fMultV0Anorm_PartLevel = 0.;
+   fMultV0Cnorm_PartLevel = 0.;
+   fMultV0AV0Cnorm_PartLevel = 0.;
+
+
    if(fMC){
+      TClonesArray* arrayMC = 0; // array particles in the MC event
+      arrayMC = (TClonesArray*) InputEvent()->FindListObject(AliAODMCParticle::StdBranchName());
+      if(!arrayMC){
+         AliError("No MC array found!");
+         return kFALSE;
+      }
+      AliAODMCParticle* daughtermc; 
```

Comment:
I recommend to use the AliParticleContainer, in particular for the Photon double counting bug you cannot rely that the particles are in the standard branch.

Reply:
Hi Markus,
in the standard branch i am looking just for the daughters of the photon. If the daughter is another photon, i discard such particle.     Or should i apply some standard approach how to filter them out?
In such a case could you point me to the piece of code where this is handeled

Many thanks
Filip

Reply:
Hi @fkrizek ,

The code is OK, but in principle we have a preparation task doing this called AliEmcalMCTrackSelector (https://github.com/alisw/AliPhysics/blob/master/PWG/EMCAL/EMCALtasks/AliEmcalMCTrackSelector.h). This task however doesn't necessarily write into the standard branch. The result from the track selection is interfaced via the AliMCParticleContainer. I recommend using standard tools.

In the mean time I will approve the PR.

Cheers

Markus